compiler/ballerina-lang/src/main/java/io/ballerina/projects/JarResolver.java

```diff
@@ -131,8 +131,14 @@ private void addPlatformLibraryPaths(PackageContext packageContext,
         // Add all the jar library dependencies of current package (packageId)
         Collection<PlatformLibrary> otherJarDependencies = jBalBackend.platformLibraryDependencies(
                 packageContext.packageId(), scope);
+
```

Comment:
IMO, the better solution would be to make a `jarFiles` a `Set` and override the `equals` method of `JarLibrary`

Reply:
@sameerajayasoma is this ok to be addressed in a PR later? RC1 is waiting on this.

Reply:
@sameerajayasoma would it be fine to address this in a PR later. RC1 is waiting on this PR.